File: 132Pattern2.py
class Solution:
    # Time Limit Exceeded
    def find132pattern(self, nums) -> bool:
        n = len(nums)
        mn = float('inf')
        for j in range(n):
            mn = min(mn, nums[j])
            if nums[j] == mn:
                continue
            for k in range(n - 1, j, -1):
                if mn < nums[k] < nums[j]:
                    return True
        return False


if __name__ == '__main__':
    nums = [int(num) for num in input().split()]
    solution = Solution()
    print(solution.find132pattern(nums))

File: 132Pattern2.cpp
#include<iostream>
#include<vector>
#include<algorithm>

using namespace std;

class Solution {
public:
    bool find132pattern(vector<int>& nums) {
    	int mn = INT_MAX, n = nums.size();
    	for(int j = 0; j < n; j++) {
    		mn = min(mn, nums[j]);
    		if(mn == nums[j]) continue;
    		for(int i = n - 1; i > j; i--) {
    			if(nums[i] > mn && nums[i] < nums[j]) return true;
			}
		}
		return false;
    }
};

int main() {
	
	int n;
	cin>>n;
	
	vector<int> nums;
	for(int i = 0; i < n; i++) {
		int num;
		cin>>num;
		nums.push_back(num);
	}
	
	Solution *solution = new Solution();
	solution->find132pattern(nums) ? cout<<"true" : cout<<"false";
	
	return 0;
	
}
